homeassistant/components/switch/broadlink.py

```diff
@@ -54,7 +54,25 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     try:
         broadlink_device.auth()
     except socket.timeout:
-        pass
+        _LOGGER.error("Failed to connect to device.")
+
+    def _learn_command(call):
+        try:
+            broadlink_device.auth()
+        except socket.timeout:
+            _LOGGER.error("Failed to connect to device.")
+            return
+        broadlink_device.enter_learning()
+        _LOGGER.info("Press the key you want HASS to learn")
+        start_time = utcnow()
+        while (utcnow() - start_time) < timedelta(seconds=20):
```

Comment:
There is still a blocking issue here. 
@balloob , @pvizeli : Any tips of how I can solve this with async?